device/sycl/include/traccc/sycl/clusterization/measurement_sorting_algorithm.hpp

```diff
@@ -21,6 +21,10 @@
 // System include(s).
 #include <functional>
 
+namespace traccc {
+struct memory_resource;
```

Comment:
No need to be too clever here. Let's just include the appropriate header for this type.